packages/node/src/nodeClient.ts

```diff
@@ -46,35 +47,50 @@ export class NodeClient implements Client<Options> {
     // Check if there's 0 events, flush is not needed.
     const arrayLength = this._events.length;
     if (arrayLength === 0) {
-      return { status: Status.Success, statusCode: 200 };
+      return SKIPPED_RESPONSE;
+    }
+
+    // Reset the response listeners and pull them out.
+    const responseListeners = this._responseListeners;
+    this._responseListeners = [];
+
+    try {
+      const eventsToSend = this._events.splice(0, arrayLength);
```

Comment:
Thoughts on treating `eventsToSend` like how `responseListeners` is handled? I think it's more descriptive than `splice`

```js
const eventsToSend = this._events;
this._events = [];
```

Reply:
oeuf, I forgot to get this from the previous PR. fixed.